devicetypes/smartthings/arrival-sensor-ha.src/arrival-sensor-ha.groovy

```diff
@@ -107,11 +144,42 @@ private handleBatteryEvent(volts) {
         log.debug "Ignoring invalid value for voltage (${volts/10}V)"
     }
     else {
-        def batteryMap = [28:100, 27:100, 26:100, 25:90, 24:90, 23:70,
+    	def batteryMap = [28:100, 27:100, 26:100, 25:90, 24:90, 23:70,
                           22:70, 21:50, 20:50, 19:30, 18:30, 17:15, 16:1, 15:0]
         def minVolts = 15
         def maxVolts = 28
+        
+        if (volts < minVolts)
+            volts = minVolts
+        else if (volts > maxVolts)
+            volts = maxVolts
+        def value = batteryMap[volts]
+        if (value != null) {
+            def linkText = getLinkText(device)
+            descriptionText = '{{ linkText }} battery was {{ value }}'
+            def eventMap = [
+                name: 'battery',
+                value: value,
+                descriptionText: descriptionText,
+                translatable: true
+            ]
+            log.debug "Creating battery event for voltage=${volts/10}V: ${linkText} ${eventMap.name} is ${eventMap.value}%"
+            sendEvent(eventMap)
+        }
+    }
+}
 
+private handleBatteryEventVision(volts) {
```

Comment:
Can You explain why did You add another (similar) method to handle battery events ?
Since the main difference is the batteryMap You may also use the handleBatteryEvent only and get battery map from a method that returns different battery maps depending on the device.

Reply:
OK! I will add a method that returns different battery maps depending on the device.